tests/test_decoders.py

```diff
@@ -214,125 +213,55 @@ async def iterator() -> typing.AsyncIterator[bytes]:
+    response = httpx.Response(
+        200, content=[b"", b"12345\r", b"foo ", b"bar ", b"baz\r"]
+    )
+    assert list(response.iter_lines()) == ["12345\n", "foo bar baz\n"]
 
 
 def test_line_decoder_crnl():
-    decoder = LineDecoder()
-    assert decoder.decode("") == []
-    assert decoder.decode("a\r\n\r\nb\r\nc") == ["a\n", "\n", "b\n"]
-    assert decoder.flush() == ["c"]
-
-    decoder = LineDecoder()
-    assert decoder.decode("") == []
-    assert decoder.decode("a\r\n\r\nb\r\nc\r\n") == ["a\n", "\n", "b\n", "c\n"]
-    assert decoder.flush() == []
-
-    decoder = LineDecoder()
-    assert decoder.decode("") == []
-    assert decoder.decode("a\r") == []
-    assert decoder.decode("\n\r\nb\r\nc") == ["a\n", "\n", "b\n"]
-    assert decoder.flush() == ["c"]
+    response = httpx.Response(200, content=[b"", b"a\r\n\r\nb\r\nc"])
+    assert list(response.iter_lines()) == ["a\n", "\n", "b\n", "c"]
+
+    response = httpx.Response(200, content=[b"", b"a\r\n\r\nb\r\nc\r\n"])
+    assert list(response.iter_lines()) == ["a\n", "\n", "b\n", "c\n"]
+
+    response = httpx.Response(200, content=[b"", b"a\r", b"\n\r\nb\r\nc"])
+    assert list(response.iter_lines()) == ["a\n", "\n", "b\n", "c"]
 
     # Issue #1033
-    decoder = LineDecoder()
-    assert decoder.decode("") == []
-    assert decoder.decode("12345\r\n") == ["12345\n"]
-    assert decoder.decode("foo ") == []
-    assert decoder.decode("bar ") == []
-    assert decoder.decode("baz\r\n") == ["foo bar baz\n"]
-    assert decoder.flush() == []
-
-
-def test_byte_chunker():
-    decoder = ByteChunker()
-    assert decoder.decode(b"1234567") == [b"1234567"]
-    assert decoder.decode(b"89") == [b"89"]
-    assert decoder.flush() == []
-
-    decoder = ByteChunker(chunk_size=3)
-    assert decoder.decode(b"1234567") == [b"123", b"456"]
-    assert decoder.decode(b"89") == [b"789"]
-    assert decoder.flush() == []
-
-    decoder = ByteChunker(chunk_size=3)
-    assert decoder.decode(b"123456") == [b"123", b"456"]
-    assert decoder.decode(b"789") == [b"789"]
-    assert decoder.flush() == []
-
-    decoder = ByteChunker(chunk_size=3)
-    assert decoder.decode(b"123456") == [b"123", b"456"]
-    assert decoder.decode(b"78") == []
-    assert decoder.flush() == [b"78"]
-
-
-def test_text_chunker():
-    decoder = TextChunker()
-    assert decoder.decode("1234567") == ["1234567"]
-    assert decoder.decode("89") == ["89"]
-    assert decoder.flush() == []
-
-    decoder = TextChunker(chunk_size=3)
-    assert decoder.decode("1234567") == ["123", "456"]
-    assert decoder.decode("89") == ["789"]
-    assert decoder.flush() == []
-
-    decoder = TextChunker(chunk_size=3)
-    assert decoder.decode("123456") == ["123", "456"]
-    assert decoder.decode("789") == ["789"]
-    assert decoder.flush() == []
-
-    decoder = TextChunker(chunk_size=3)
-    assert decoder.decode("123456") == ["123", "456"]
-    assert decoder.decode("78") == []
-    assert decoder.flush() == ["78"]
```

Comment:
Most of these cases we actually already well covered in `test_responses.py`, so I figured I'd expand those a bit to match coverage, and drop these.